modules/evs/ut-coverage/evs_UT.c

```diff
@@ -1814,6 +1862,126 @@ void Test_InvalidCmd(void)
+        InjectedTime = OS_TimeAssembleFromMilliseconds(0, 1000 * CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST /
+                                                                  CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC +
+                                                              1000 / CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC);
+        UT_SetDataBuffer(UT_KEY(CFE_PSP_GetTime), &InjectedTime, sizeof(InjectedTime), false);
+        EVS_Retval = SendEventFuncs[j](EVENT_ID);
+        UtAssert_UINT32_EQ(SnapshotData.Count, CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST + 2);
+        UtAssert_UINT32_EQ(CapturedTlm.Payload.PacketID.EventID, EVENT_ID);
+        UtAssert_UINT32_EQ(EVS_Retval, CFE_SUCCESS);
+
+        /*
+         * Test re-squelching event message
+         */
+        UT_SetDataBuffer(UT_KEY(CFE_PSP_GetTime), &InjectedTime, sizeof(InjectedTime), false);
+        EVS_Retval = SendEventFuncs[j](EVENT_ID);
+        UtAssert_UINT32_EQ(SnapshotData.Count, CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST + 3);
+        UtAssert_UINT32_EQ(CapturedTlm.Payload.PacketID.EventID, CFE_EVS_SQUELCHED_ERR_EID);
+        UtAssert_UINT32_EQ(EVS_Retval, CFE_EVS_APP_SQUELCHED);
+
+        /*
+         * Test re-squelching
+         */
+        UT_SetDataBuffer(UT_KEY(CFE_PSP_GetTime), &InjectedTime, sizeof(InjectedTime), false);
+        EVS_Retval = SendEventFuncs[j](EVENT_ID);
+        UtAssert_UINT32_EQ(SnapshotData.Count, CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST + 3);
+        UtAssert_UINT32_EQ(EVS_Retval, CFE_EVS_APP_SQUELCHED);
+
+        /*
+         * Test positive saturation of tokens
+         */
+        InjectedTime = OS_TimeAssembleFromMilliseconds(0, 3 * 1000 * CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST /
+                                                                  CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC +
+                                                              1000 / CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC);
+        UT_SetDataBuffer(UT_KEY(CFE_PSP_GetTime), &InjectedTime, sizeof(InjectedTime), false);
+        EVS_Retval = SendEventFuncs[j](EVENT_ID);
+        UtAssert_UINT32_EQ(SnapshotData.Count, CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST + 4);
+        UtAssert_UINT32_EQ(CapturedTlm.Payload.PacketID.EventID, EVENT_ID);
+        UtAssert_UINT32_EQ(EVS_Retval, CFE_SUCCESS);
+        UtAssert_UINT32_EQ(AppDataPtr->SquelchedCount, CFE_EVS_MAX_SQUELCH_COUNT);
+
+        /*
+         * Test boundary condition where SquelchTokens > -EVENT_COST && CreditCount >= EVENT_COST
+         */
+        InjectedTime = OS_TimeAssembleFromMilliseconds(1, 150);
+        UT_SetDataBuffer(UT_KEY(CFE_PSP_GetTime), &InjectedTime, sizeof(InjectedTime), false);
+        AppDataPtr->SquelchedCount            = 0;
+        AppDataPtr->SquelchTokens             = -1500;
+        AppDataPtr->LastSquelchCreditableTime = OS_TimeAssembleFromMilliseconds(1, 0);
+        EVS_Retval                            = SendEventFuncs[j](EVENT_ID);
+        UtAssert_UINT32_EQ(EVS_Retval, CFE_EVS_APP_SQUELCHED);
```

Comment:
```
If CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC == 8:
    150ms * 8 == positive credit count of 1200 which is < 1500
    test passes
however, if CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC == 16:
    150ms * 16 == positive credit count of 2400 which is > 1500
    test fails
 ```   
 Injected time difference should be 1200/CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC  rather than hardcoded 150ms to account for varying values of CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC (up to a point where CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC happens to be greater than an unreasonable 1200 and breaks the test again)

Reply:
Agreed. Will proceed with unit test update.